solidity-v1/dashboard/src/hooks/useAutoConnect.js

```diff
@@ -41,29 +41,56 @@ const useAutoConnect = () => {
 
   useEffect(() => {
     if (injectedTried) return
-    injected.getAccounts().then((accounts) => {
-      setInjectedTried(true)
-      if (
-        (!isEmptyArray(accounts) && isExactRoutePath) ||
-        isWalletFromUrlSameAsInMetamask(accounts)
-      ) {
-        connectAppWithWallet(injected, false).catch((error) => {
-          // Just log an error, we don't want to do anything else.
-          console.log(
-            "Eager injected connector cannot connect with the dapp:",
-            error.message
-          )
-        })
-      } else if (walletAddressFromUrl && !connector) {
-        const explorerModeConnector = new ExplorerModeConnector()
-        openModal(MODAL_TYPES.ExplorerMode, {
-          connectAppWithWallet,
-          connector: explorerModeConnector,
-          address: walletAddressFromUrl,
-          connectWithWalletOnMount: true,
-        })
-      }
-    })
+
+    if (window.tally) {
+        tallyInjectedConnector.getAccounts().then((accounts) => {
+        setInjectedTried(true)
+        if (
+          (!isEmptyArray(accounts) && isExactRoutePath) ||
+          isWalletFromUrlSameAsInMetamask(accounts)
+        ) {
+          connectAppWithWallet(tallyInjectedConnector, false).catch((error) => {
+            // Just log an error, we don't want to do anything else.
+            console.log(
+              "Eager injected connector cannot connect with the dapp:",
+              error.message
+            )
+          })
+        } else if (walletAddressFromUrl && !connector) {
+          const explorerModeConnector = new ExplorerModeConnector()
+          openModal(MODAL_TYPES.ExplorerMode, {
+            connectAppWithWallet,
+            connector: explorerModeConnector,
+            address: walletAddressFromUrl,
+            connectWithWalletOnMount: true,
+          })
+        }
+      })
```

Comment:
In both ifs we have basically the same code but with different connector. We should extract if to a separate method that receive connector as an argument. For example:

`const connectWallet = (connector) = (...) `

and then just:
```suggestion
connectWallet(tallyInjectedConnector)
```

Reply:
https://github.com/keep-network/keep-core/pull/2744/commits/0c22c3d078d822867fc8de0c83eeeab309a01192